scripts/set-environment

```diff
@@ -76,18 +81,25 @@ echo
 # Avoid ssh prompting when connecting to new ssh hosts
 mkdir -p $HOME/.ssh && echo "StrictHostKeyChecking no" >> "$HOME/.ssh/config"
 
+# Create the artifacts directory if needed
+if [ ! -d "$ARTIFACTS_FULL_DIR" ]; then
+  mkdir -p $ARTIFACTS_FULL_DIR
+fi
+
 # Configure the GitHub Oauth token if it is available
-if [ -n "$GITHUB_TOKEN" ] ; then
+if [ -n "$GITHUB_TOKEN" ] && [ -x "$(command -v composer)" ]; then
```

Comment:
This check doesn't hurt, but Composer should always be installed in this image (we install it in the Dockerfile)

Reply:
My thought is to copy this script to cache in an early job on CircleCI, then source it in each subsequent job. While this container has Composer the visual regression container, for example, does not.

Reply:
Can we just cache `$BASH_ENV` instead? While other containers might need some of the other services provided by this script, it seems error-prone to me to maintain a script that will run in multiple environments. `$BASH_ENV` is safe for this, because it only sets variables. The risk is that someone might add stuff to this script later not realizing its effect on visual regression.

Reply:
> Can we just cache `$BASH_ENV` instead

If we cache the result of `$BASH_ENV` and source that in each job the rest of the script, such as setting the Git token for Composer, won't run in those jobs.

Reply:
Yeah, I think those small tasks should be duplicated in the container that needs the service.

Reply:
The git token for Composer is only needed for job steps that run `composer require` and similar, for example; we should strive to avoid doing more installation in later jobs, and instead pass build results along in the cache.

Reply:
I'm not sure if that gets the stuff we need to duplicate in other containers down to the empty set or not, but there's only a few things the script does right now, so I think we can pare it down to close to zero.